extensions/tika/runtime/src/main/java/io/quarkus/tika/TikaParser.java

```diff
@@ -0,0 +1,72 @@
+package io.quarkus.tika;
+
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.tika.io.TikaInputStream;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.parser.AutoDetectParser;
+import org.apache.tika.parser.ParseContext;
+import org.apache.tika.parser.Parser;
+import org.apache.tika.sax.ToTextContentHandler;
+import org.jboss.logging.Logger;
+import org.xml.sax.ContentHandler;
+import org.xml.sax.helpers.DefaultHandler;
+
+public class TikaParser {
```

Comment:
I think this should be an `@ApplicationScoped` bean that we can inject.

The Tika parser is supposed to be thread safe so it would be highly beneficial if it was instantiated only once.